### PR TITLE
(MVP) User can choose origin or destination for minischedule trip labels

### PR DESCRIPTION
**Pull request by phildarnowsky**
Card: https://app.asana.com/0/1148853526253426/1199683601247945

We've had a number of requests to make this configurable, this is an MVP behind a feature flag until the design is finalized.

### CODE REVIEW

**Comment by phildarnowsky** [comment]
Note to self: assuming we do proceed with this ticket, remember to hardcode a translation for `dudly`

File: assets/src/components/settingsPage.tsx [hunk 1]
```diff
@@ -53,6 +57,32 @@ const SettingsPage = (): ReactElement<HTMLDivElement> => {
...
+            <DropdownSetting
+              icon={mapIcon}
+              label="Trip Label"
+              selectId="minischedules-trip-label-setting"
+              value={userSettings.minischedulesTripLabel}
+              onChange={
+                // Ignoring this because we can't get the tests working and
+                // neither of can figure out why
+                /* istanbul ignore next */
+                (value) => {
```
Comment (lemald) [inline assets/src/components/settingsPage.tsx:71]: Given what I learned in the process of working on #1035, do you think you could use that same basic approach to get tests for this working?

File: lib/schedule/data.ex [hunk 2]
```diff
@@ -269,6 +269,7 @@ defmodule Schedule.Data do
     gtfs_trips = Gtfs.Trip.parse(gtfs_files["trips.txt"], bus_route_ids)
     gtfs_trip_ids = MapSet.new(gtfs_trips, & &1.id)
     stop_times_by_id = StopTime.parse(gtfs_files["stop_times.txt"], gtfs_trip_ids)
+
```
Comment (lemald) [inline lib/schedule/data.ex:272]: Diff seems extraneous to the rest of the PR, may have been committed accidentally but I'm not sure how much removing it matters.

File: assets/src/userSettings.ts [hunk 3]
```diff
@@ -25,11 +32,16 @@ export const vehicleLabelSetting = (
...
+export type VehicleLabelData = "run_id" | "vehicle_id"
+export type TripLabelData = "origin" | "destination"
 
 interface SettingsData {
   ladder_page_vehicle_label: VehicleLabelData
   shuttle_page_vehicle_label: VehicleLabelData
+  // We allow minischedules_trip_label to be undefined to handle the
+  // hybrid code problem, we can disallow undefined after the first
+  // deploy of the corresponding backend code.
+  minischedules_trip_label?: TripLabelData
```
Comment (lemald) [inline assets/src/userSettings.ts:44]: Ah, this makes me realize that I should do the same in #1035 before merging.

Comment (lemald) [reply assets/src/userSettings.ts:44]: Though once both of these are deployed, it's probably worth cleaning up the front-end code.

File: lib/schedule/data.ex (hunk 2 above)
Comment (phildarnowsky) [reply lib/schedule/data.ex:272]: eh, it takes one extra second to remove it, and it keeps the commit history that much cleaner

File: assets/src/components/settingsPage.tsx (hunk 1 above)
Comment (phildarnowsky) [reply assets/src/components/settingsPage.tsx:71]: I'll give it a whirl

File: assets/src/userSettings.ts (hunk 3 above)
Comment (phildarnowsky) [reply assets/src/userSettings.ts:44]: yes we should make tickets to remove these bits at the first convenient opportunity (post-deploy)

Comment (lemald) [reply assets/src/userSettings.ts:44]: I created [a ticket](https://app.asana.com/0/1148853526253420/1199999682730453/f).

**Comment by phildarnowsky** [comment]
@lemald ready for another look

**Review by lemald — APPROVED** [review]
Looks good to me! Just let me know when you're done rebasing this after I merge #1035 and I can take one last look.

**Pull request merged**